lib/salus/report.rb

```diff
@@ -187,9 +136,44 @@ def export_report
 
     private
 
-    # Wrap lines to 100 chars by default
-    def wrap(text, line_width = 100)
-      text.gsub(/(.{1,#{line_width}})/, "\\1\n")
+    def deep_stringify_keys(datum)
+      case datum
+      when Hash then datum.map { |key, value| [key.to_s, deep_stringify_keys(value)] }.to_h
+      when Array then datum.map { |value| deep_stringify_keys(value) }
+      else datum
+      end
+    end
+
+    # FIXME(as3richa): factor this out along with the identical version
+    # in scan_report.rb. Maybe have a Formatting mixin module for wrapping,
+    # indentation, coloring, tabulation...
+    def wrapify(string, wrap)
+      return string if wrap.nil?
+
+      parts = []
+
+      string.each_line("\n").each do |line|
+        if line == "\n"
+          parts << "\n"
+          next
+        end
+
+        line = line.chomp
+        index = 0
+
+        while index < line.length
+          parts << line.slice(index, wrap) + "\n"
+          index += wrap
+        end
+      end
+
+      parts.join
+    end
+
+    def indent(text)
+      # each_line("\n") rather than split("\n") because the latter
+      # discards trailing empty lines. Also, don't indent empty lines
+      text.each_line("\n").map { |line| line == "\n" ? "\n" : ("  " + line) }.join
```

Comment:
would make the string `"  "` a constant at the top or make the constant `INDENT_SIZE = 2` and then to `' ' * INDENT_SIZE` to get the string.

nit: also should use single quotes for the indent string/space.